Kconfig.zephyr

```diff
@@ -512,9 +512,35 @@ config BUILD_OUTPUT_META
 	  Create a build meta file in the build directory containing lists of:
 	  - Zephyr: path and revision (if git repo)
 	  - Zephyr modules: name, path, and revision (if git repo)
-	  - West projects: path and revision
+	  - West:
+	    - manifest: path and revision
+	    - projects: path and revision
+	  - Workspace:
+	    - dirty: one or more repositories are marked dirty
+	    - extra: extra Zephyr modules are manually included in the build
+	    - off:   the SHA of one or more west projects has diverged since the
+	             last `west update`.
```

Comment:
> I think this help section has been trying very hard to avoid mentioning the term manifest-rev and provide some user-friendlier description instead, but that effort has failed.

I really agree with this

Reply:
> avoid mentioning the term manifest-rev and provide some user-friendlier description instead

Actually, the two are not mutually exclusive. A very short and semi-accurate description may not hurt as long as the handle to the complete, accurate and unambiguous definition is there too.

Reply:
> I think this help section has been trying very hard to avoid mentioning the term manifest-rev 

No that was not the reason.
The main reason was that this is already a pretty long Kconfig help text, and therefore I try to just have a brief description for each state.

I took @mbolivar-nordic proposal from here: https://github.com/zephyrproject-rtos/zephyr/pull/40085#discussion_r753616469 but left out the part inside the parenthesis.
Feel free to rewrite to use the `manifest-rev` directly if you prefer, but I felt the way it was split in @mbolivar-nordic proposal with text and extra parenthesis sentence was not needed for a brief help text.

Reply:
```
	    - off:   the SHA of one or more west projects are not what the
	               manifest defined when `west update` was run the last
                       time (`manifest-rev`).
```
Any sentence really as long as it includes the critical keyword `manifest-rev` that saves a longer description. This suggestion for instance omits the `west update proj1 proj2` possibility; it may miss other corner cases too.

Reply:
@marc-hb thanks.

Kconfig help updated to proposed text.
(and commit messages has detailed state description removed to avoid them getting out-of-sync :wink:  )